include/mgp_mock.py

```diff
@@ -0,0 +1,1011 @@
+            self._path = starting_vertex_or_path
+
+        elif isinstance(starting_vertex_or_path, Vertex):
+            # for consistency with the Python API, the `_vertex` attribute isn’t a “public” property
+            vertex = starting_vertex_or_path._vertex
+            if not vertex.is_valid():
+                raise InvalidContextError()
+
+            self._path = _mgp_mock.Path.make_with_start(vertex, vertex.graph)
+
+        else:
+            raise TypeError(f"Expected 'Vertex' or '_mgp_mock.Path', got '{type(starting_vertex_or_path)}'")
+
+        self._vertices = None
+        self._edges = None
+
+    def __copy__(self):
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        assert len(self.vertices) >= 1
+
+        path = Path(self.vertices[0])
+        for e in self.edges:
+            path.expand(e)
+
+        return path
+
+    def __deepcopy__(self, memo):
+        try:
+            return Path(memo[id(self._path)])
+        except KeyError:
+            pass
+        # This is the same as the shallow copy, as the underlying C API should
+        # not support deepcopy. Besides, it doesn't make much sense to actually
+        # copy _mgp_mock.Edge and _mgp_mock.Vertex types as they are actually references
+        # to graph elements and not proper values.
+        path = self.__copy__()
+        memo[id(self._path)] = path._path
+        return path
+
+    def is_valid(self) -> bool:
+        return self._path.is_valid()
+
+    def expand(self, edge: Edge):
+        if not isinstance(edge, Edge):
+            raise TypeError(f"Expected 'Edge', got '{type(edge)}'")
+
+        if not self.is_valid() or not edge.is_valid():
+            raise InvalidContextError()
+
+        # for consistency with the Python API, the `_edge` attribute isn’t a “public” property
+        self._path.expand(edge._edge)
+
+        self._vertices = None
+        self._edges = None
+
+    @property
+    def vertices(self) -> typing.Tuple[Vertex, ...]:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        if self._vertices is None:
+            num_vertices = self._path.size() + 1
+            self._vertices = tuple(Vertex(self._path.vertex_at(i)) for i in range(num_vertices))
+
+        return self._vertices
+
+    @property
+    def edges(self) -> typing.Tuple[Edge, ...]:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        if self._edges is None:
+            num_edges = self._path.size()
+            self._edges = tuple(Edge(self._path.edge_at(i)) for i in range(num_edges))
+
+        return self._edges
+
+
+class Vertices:
+    __slots__ = ("_graph", "_len")
+
+    def __init__(self, graph):
+        if not isinstance(graph, nx.MultiDiGraph):
+            raise TypeError(f"Expected 'networkx.classes.multidigraph.MultiDiGraph', got '{type(graph)}'")
+
+        self._graph = graph
+        self._len = None
```

Comment:
Why does Vertices not accept _mgp_mock.Graph? I think there is no need for nx.MultiDiGraph.

Reply:
Changed in the meantime.